src/contexts/Prices/PricesProvider.tsx

```diff
@@ -67,32 +83,74 @@ const PricesProvider: React.FC = ({ children }) => {
   }, [])
 
   useEffect(() => {
-    if (!indexPrice || !usdInEthDpiPool) return
+    if (
+      !indexPrice ||
+      !usdInEthDpiPool ||
+      !totalSupplyInEthDpiPool ||
+      !ethereum ||
+      !farmTwoAddress
+    )
+      return
 
     const totalTokenEmissionsPerDay = 700
     const totalUSDEmissionPerDay =
       totalTokenEmissionsPerDay * Number(indexPrice)
-    const dailyYield = new BigNumber(totalUSDEmissionPerDay)
-      .dividedBy(new BigNumber(usdInEthDpiPool))
-      .multipliedBy(100)
-    const calculatedApy = dailyYield.multipliedBy(365)
 
-    setFarmTwoApy(calculatedApy.toFixed(2))
-  }, [usdInEthDpiPool, indexPrice])
+    //get usd price per lp token
+    const pricePerLPToken = new BigNumber(usdInEthDpiPool).dividedBy(
+      new BigNumber(totalSupplyInEthDpiPool)
+    )
+
+    //multiply by totalSupply
+    getAmountOfStakedTokens(ethereum, farmTwoAddress).then(
+      (tokensInStakingContract) => {
+        const usdInStakingContract = new BigNumber(tokensInStakingContract)
+          .dividedBy(new BigNumber(10).pow(18))
+          .multipliedBy(pricePerLPToken)
+        const dailyYield = new BigNumber(totalUSDEmissionPerDay)
+          .dividedBy(usdInStakingContract)
+          .multipliedBy(100)
+        const calculatedApy = dailyYield.multipliedBy(365)
+
+        setFarmTwoApy(calculatedApy.toFixed(2))
+      }
+    )
+  }, [usdInEthDpiPool, indexPrice, ethereum, totalSupplyInEthDpiPool])
 
   useEffect(() => {
-    if (!indexPrice || !usdInEthMviPool) return
+    if (
+      !indexPrice ||
+      !usdInEthMviPool ||
+      !totalSupplyInEthMviPool ||
+      !ethereum ||
+      !mviStakingRewardsAddress
+    )
+      return
 
     const totalTokenEmissionsPerDay = 127
     const totalUSDEmissionPerDay =
       totalTokenEmissionsPerDay * Number(indexPrice)
-    const dailyYield = new BigNumber(totalUSDEmissionPerDay)
-      .dividedBy(new BigNumber(usdInEthMviPool))
-      .multipliedBy(100)
-    const calculatedApy = dailyYield.multipliedBy(365)
 
-    setMviRewardsApy(calculatedApy.toFixed(2))
-  }, [usdInEthMviPool, indexPrice])
+    //get usd price per lp token
+    const pricePerLPToken = new BigNumber(usdInEthMviPool).dividedBy(
+      new BigNumber(totalSupplyInEthMviPool)
+    )
+
+    //multiply by totalSupply
+    getAmountOfStakedTokens(ethereum, mviStakingRewardsAddress).then(
```

Comment:
Are there any problems if `getAmountOfStakedTokens` request fails? Do we need to catch anything?

Reply:
Great catch, I'll provide some error handling today. There's not a lot we can do data-wise if that happens, but we can zero out everything and log an error.